base_preload_compare/models/base_model.py

```diff
@@ -88,9 +88,15 @@ def _check_any_changes(self, fields, line):
 
     def _check_preload_compare(self):
         """ Check if we are installing/upgrading and if Model-Module combination is marked for preload """
+        bpc_module = self.env['ir.module.module'].search([
+            ('name', '=', 'base_preload_compare'),
+        ])
+        if not bpc_module.state == 'installed':
```

Comment:
What about checking the registry only?
```suggestion
        if not self.env.get('base.preload.compare')
```

Reply:
Thanks for the tip, changed :)